src/test/ui/associated-type-bounds/bounds-on-assoc-in-trait.rs

```diff
@@ -18,6 +18,7 @@ impl<'a, 'b> Lam<&'a &'b u8> for L2 { type App = u8; }
 
 trait Case1 {
     type A: Iterator<Item: Debug>;
+    //~^ ERROR `<<Self as Case1>::A as std::iter::Iterator>::Item` doesn't implement `std::fmt::Debug`
```

Comment:
Isn't this technically a regression? (For the same reason https://github.com/rust-lang/rust/pull/69741 might not land.)

Reply:
This is an unstable feature that doesn't work correctly in this position with or without this change.